pkg/reconciler/delivery/delivery.go

```diff
@@ -131,21 +181,37 @@ func modifyRouteSpec(route *v1.Route, newRevName string) (*v1.Route, error) {
 				Percent: ptr.Int64(100),
 			},
 		}
+		return route, nil
 	} else if len(route.Status.Traffic) == 1 {
```

Comment:
no need for `else if`, since there's a `return` in the `if`
Basically
```
if a {
 ...
 return
}
if b {
 ...
 return
}
...
```

Reply:
fixed